src/Android/Xamarin.Android/Samples/Network Analysis/NavigateRoute/NavigateRoute.cs

```diff
@@ -0,0 +1,294 @@
+        }
+
+        private async void Initialize()
+        {
+            try
+            {
+                // Create the map view.
+                _myMapView.Map = new Map(Basemap.CreateNavigationVector());
+
+                // Create the text to speech object.
+                _textToSpeech = new TextToSpeech(this, this, "com.google.android.tts");
+
+                // Create the route task, using the online routing service.
+                RouteTask routeTask = await RouteTask.CreateAsync(_routingUri);
+
+                // Get the default route parameters.
+                RouteParameters routeParams = await routeTask.CreateDefaultParametersAsync();
+
+                // Explicitly set values for parameters.
+                routeParams.ReturnDirections = true;
+                routeParams.ReturnStops = true;
+                routeParams.ReturnRoutes = true;
+                routeParams.OutputSpatialReference = SpatialReferences.Wgs84;
+
+                // Create a Stop for first location.
+                Stop stop1 = new Stop(_startLocation);
+
+                // Create a Stop for the destination (RH Fleet Aerospace Museum).
+                Stop stop2 = new Stop(_destinationLocation);
+
+                // Assign the stops to the route parameters.
+                List<Stop> stopPoints = new List<Stop> { stop1, stop2 };
+                routeParams.SetStops(stopPoints);
+
+                // Get the route results.
+                _routeResult = await routeTask.SolveRouteAsync(routeParams);
+                _route = _routeResult.Routes[0];
+
+                // Add a graphics overlay for the route graphics.
+                _myMapView.GraphicsOverlays.Add(new GraphicsOverlay());
+
+                // Add graphics for the start and end points.
+                _myMapView.GraphicsOverlays[0].Graphics.Add(new Graphic(_startLocation, new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, Color.Green, 25)));
+                _myMapView.GraphicsOverlays[0].Graphics.Add(new Graphic(_destinationLocation, new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.X, Color.Red, 25)));
+
+                // Create a graphic (with a dashed line symbol) to represent the route.
+                _routeAheadGraphic = new Graphic(_route.RouteGeometry) { Symbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.BlueViolet, 5) };
+
+                // Create a graphic (solid) to represent the route that's been traveled (initially empty).
+                _routeTraveledGraphic = new Graphic { Symbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.LightBlue, 3) };
+
+                // Add the route graphics to the map view.
+                _myMapView.GraphicsOverlays[0].Graphics.Add(_routeAheadGraphic);
+                _myMapView.GraphicsOverlays[0].Graphics.Add(_routeTraveledGraphic);
+
+                // Set the map viewpoint to show the entire route.
+                await _myMapView.SetViewpointGeometryAsync(_route.RouteGeometry, 100);
```

Comment:
I recommend having the navigation button disabled at startup, then only enable it once the sample is fully ready.